### ISSUE TITLE
PLW0602 false positive

### ISSUE THREAD

**Issue by rbubley** [astral-sh/ruff#10078]
```python
a = {}


def f() -> None:
    global a
    a[1] = 1
    a[2] = 2


if __name__ == "__main__":
    f()
    print(a)
```
Command: `ruff  check --isolated  --select PLW0602 file.py `

Ruff output:
```
file.py:5:12: PLW0602 Using global for `a` but no assignment is done
Found 1 error.
```

Ruff version: 0.2.2

**Comment by graingert** [comment]
This isn't a false positive, there's indeed no assignments to a, only keys of a

**Comment by MichaReiser** [comment]
Nice catch @graingert 

That's correct. You can remove the `global a` and Python will resolve `a` to the symbol from the outer scope

```python
a = {}


def f() -> None:
    a[1] = 1
    a[2] = 2


if __name__ == "__main__":
    f()
    print(a)
```
